libs/message-coder/src/dynamic_bytes.test.ts

```diff
@@ -0,0 +1,50 @@
+import { Buffer } from 'buffer';
+import { err, ok } from '@votingworks/basics';
+import { dynamicBytes } from './dynamic_bytes';
+import { uint8 } from './uint8_coder';
+import { MAX_UINT8 } from './constants';
+
+test('canEncode', () => {
+  const coder = dynamicBytes(uint8());
+  expect(coder.canEncode(Buffer.from('hello'))).toEqual(true);
+  expect(coder.canEncode(1)).toEqual(false);
+});
+
+test('default', () => {
+  const coder = dynamicBytes(uint8());
+  expect(coder.default()).toEqual(Uint8Array.of());
+});
+
+test('bitLength', () => {
+  const coder = dynamicBytes(uint8());
+  expect(coder.bitLength(Uint8Array.of(1, 2, 3))).toEqual(ok(8 + 24));
+});
+
+test('encode', () => {
+  const coder = dynamicBytes(uint8());
```

Comment:
It's overkill to test every coder, but can we test `dynamicBytes(literal(...))`? `literal` and `uint8` are the 2 cases I've seen in the paper-handler manual.

Reply:
`literal` doesn't actually work here, but I added a couple variants to achieve what you want:

```ts
byteArray(64); // encodes/decodes 64 bytes as-is
byteArrayWithLengthPrefix(uint8()); // encodes/decodes 0-255 bytes by encoding the length followed by the bytes
byteArrayWithLengthPrefix(64); // encodes/decodes 64 bytes by encoding the (fixed) length followed by the bytes
```